test/unit/mssqlProtocolHandler.test.ts

```diff
@@ -3,119 +3,259 @@
+            appendLine: () => sinon.stub(),
+        }) as unknown as vscode.OutputChannel;
+
+        sinon.stub(mockVscodeWrapper, "outputChannel").get(() => {
+            return outputChannel;
+        });
+
+        sandbox.stub(Logger, "create").returns(mockLogger);
+
         sqlToolsServiceClientMock = TypeMoq.Mock.ofType(
             SqlToolsServiceClient,
             TypeMoq.MockBehavior.Loose,
         );
 
         mockGetCapabilitiesRequest(sqlToolsServiceClientMock);
 
-        mssqlProtocolHandler = new MssqlProtocolHandler(sqlToolsServiceClientMock.object);
-    });
+        mssqlProtocolHandler = new MssqlProtocolHandler(
+            mockVscodeWrapper,
+            mockMainController,
+            sqlToolsServiceClientMock.object,
+        );
 
-    test("handleUri - with no command and empty query - returns undefined", async () => {
-        const connInfo = await mssqlProtocolHandler.handleUri(
-            Uri.parse("vscode://ms-mssql.mssql/"),
+        openConnectionDialogStub = sandbox.stub(
+            // eslint-disable-next-line @typescript-eslint/no-explicit-any
+            mssqlProtocolHandler as any,
+            "openConnectionDialog",
         );
 
-        assert.isUndefined(connInfo);
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        connectProfileStub = sandbox.stub(mssqlProtocolHandler as any, "connectProfile").resolves();
     });
 
-    test("handleUri - with connect command and no query - doesn't parse query and returns undefined", async () => {
-        const connInfo = await mssqlProtocolHandler.handleUri(
-            Uri.parse("vscode://ms-mssql.mssql/connect"),
-        );
-
-        assert.isUndefined(connInfo);
+    teardown(() => {
+        sandbox.restore();
     });
 
-    test("handleUri - with connect command and connection string - parses connection string and returns connection info object", async () => {
-        const connInfo = await mssqlProtocolHandler.handleUri(
-            Uri.parse(
-                "vscode://ms-mssql.mssql/connect?connectionString=Server=myServerAddress;Database=myDataBase;User Id=myUsername;Password=myPassword;",
-            ),
-        );
+    test("No command", async () => {
+        await mssqlProtocolHandler.handleUri(Uri.parse("vscode://ms-mssql.mssql/"));
 
-        assert.isDefined(connInfo);
-        assert.equal(
-            connInfo.connectionString,
-            "Server=myServerAddress;Database=myDataBase;User Id=myUsername;Password=myPassword;",
-        );
+        expect(openConnectionDialogStub).to.have.been.calledOnceWith(undefined);
     });
 
-    test("handleUri - with connect command and query - parses query and returns connection info object", async () => {
-        let uri =
-            "vscode://ms-mssql.mssql/connect?server=myServer&database=dbName&authenticationType=SqlLogin&connectTimeout=15&trustServerCertificate=true&user=testUser";
+    suite("Connect command", () => {
```

Comment:
add test to ensure that all properties passed in the partial connection are matched